sdk/cosmos/azure-cosmos/azure/cosmos/_service_unavailable_retry_policy.py

```diff
@@ -0,0 +1,71 @@
+# The MIT License (MIT)
+# Copyright (c) Microsoft Corporation. All rights reserved.
+
+"""Internal class for service unavailable retry policy implementation in the Azure
+Cosmos database service.
+"""
+from azure.cosmos.documents import _OperationType
+
+#cspell:ignore ppaf
+
+class _ServiceUnavailableRetryPolicy(object):
+
+    def __init__(self, connection_policy, global_endpoint_manager, pk_range_wrapper, *args):
+        self.retry_after_in_milliseconds = 500
+        self.global_endpoint_manager = global_endpoint_manager
+        self.pk_range_wrapper = pk_range_wrapper
+        self.retry_count = 0
+        self.connection_policy = connection_policy
+        self.request = args[0] if args else None
+        # If an account only has 1 region, then we still want to retry once on the same region
+        self._max_retry_attempt_count = max(2, len(self.global_endpoint_manager.location_cache
+                                                   .read_regional_routing_contexts))
+        if self.request and _OperationType.IsWriteOperation(self.request.operation_type):
+            self._max_retry_attempt_count = max(2, len(
+                self.global_endpoint_manager.location_cache.write_regional_routing_contexts))
+
+    def ShouldRetry(self, _exception):
```

Comment:
Don't we need to extract the error code from `_exception` here?

Reply:
the `service_unavailable_retry_policy` is already special-cased only to 503 status codes when it gets picked as the retry policy to use:
<img width="667" height="77" alt="image" src="https://github.com/user-attachments/assets/78d6dc39-243d-4564-95af-d41c4d3a0e8b" />